modules/core/src/main/java/org/apache/ignite/internal/processors/cache/query/GridCacheDistributedQueryManager.java

```diff
@@ -198,54 +201,74 @@ protected void removeQueryFuture(long reqId) {
         assert req.mvccSnapshot() != null || !cctx.mvccEnabled() || req.cancel() ||
             (req.type() == null && !req.fields()) : req; // Last assertion means next page request.
 
-        if (req.cancel()) {
-            cancelIds.add(new CancelMessageId(req.id(), sndId));
+        boolean performanceStatsEnabled = cctx.kernalContext().performanceStatistics().enabled();
 
-            if (req.fields())
-                removeFieldsQueryResult(sndId, req.id());
-            else
-                removeQueryResult(sndId, req.id());
-        }
-        else {
-            if (!cancelIds.contains(new CancelMessageId(req.id(), sndId))) {
-                if (!F.eq(req.cacheName(), cctx.name())) {
-                    GridCacheQueryResponse res = new GridCacheQueryResponse(
-                        cctx.cacheId(),
-                        req.id(),
-                        new IgniteCheckedException("Received request for incorrect cache [expected=" + cctx.name() +
-                            ", actual=" + req.cacheName()),
-                        cctx.deploymentEnabled());
+        if (performanceStatsEnabled)
+            IoStatisticsQueryHelper.startGatheringQueryStatistics();
 
-                    sendQueryResponse(sndId, res, 0);
-                }
-                else {
-                    threads.put(req.id(), Thread.currentThread());
+        try {
+            if (req.cancel()) {
+                cancelIds.add(new CancelMessageId(req.id(), sndId));
 
-                    try {
-                        GridCacheQueryInfo info = distributedQueryInfo(sndId, req);
+                if (req.fields())
+                    removeFieldsQueryResult(sndId, req.id());
+                else
+                    removeQueryResult(sndId, req.id());
+            }
+            else {
```

Comment:
Typo: this can be rewritten as `else if (!cancelIds.contains(new CancelMessageId(req.id(), sndId))) {` to reduce one indentation level.

Reply:
Fixed.